src/classic/class/ReactClass.js

```diff
@@ -776,10 +765,10 @@ var ReactClassMixin = {
    * @deprecated
    */
   replaceProps: function(newProps, callback) {
-    ReactInstanceMap.get(this).replaceProps(
-      newProps,
-      callback
-    );
+    ReactUpdateQueue.enqueueReplaceProps(this, newProps);
+    if (callback) {
+      ReactUpdateQueue.enqueueCallback(callback);
```

Comment:
enqueueCallback takes 2 args, the first being the instance... that no tests are failing is worrisome...